frontend/app/src/components/settings/controls/SettingsOption.vue

```diff
@@ -8,14 +8,31 @@
 import { defineComponent, PropType, toRefs } from '@vue/composition-api';
 import { get, MaybeRef } from '@vueuse/core';
 import { useClearableMessages, useSettings } from '@/composables/settings';
+import { SessionState } from '@/store/session/types';
+import { useStore } from '@/store/utils';
+import { FrontendSettingsPayload } from '@/types/frontend-settings';
 import { SettingsUpdate } from '@/types/user';
 
 export default defineComponent({
   name: 'SettingsOption',
   props: {
     setting: {
       required: true,
-      type: String as PropType<keyof SettingsUpdate>
+      type: String as PropType<
+        | keyof SettingsUpdate
+        | keyof FrontendSettingsPayload
+        | keyof SessionState
```

Comment:
It would be nice if there was a way to scope it based on the other properties for auto-completion.

Also, we should probably expose only the sub-part of the `SessionState` that is changeable since not all of it should be modifiable by the user.

Reply:
I'm afraid there's no way to validate a prop based on the other prop value in vue.

The thing I can do maybe is to watch the props, and log an error if there is a mismatch between the setting location and the type.

Reply:
Yes, a validation would actually help, not with auto-completion but it will still give us way to make sure that the proper value is passed.

If it is not too much to do can you take care of it?

Also is the limiting of the session state something easy to do? (I was thinking of using typescript's standard library to create a sub-set of the keys.)

Reply:
alright it won't be that hard